src/pybind/mgr/cephadm/services/ingress.py

```diff
@@ -89,7 +89,23 @@ def haproxy_generate_config(
 
         if backend_spec.service_type == 'nfs':
             mode = 'tcp'
-            by_rank = {d.rank: d for d in daemons if d.rank is not None}
+            ranked_daemons = [d for d in daemons if (d.rank is not None and d.rank_generation is not None)]
+            by_rank: Dict[int, DaemonDescription] = {}
+            for d in ranked_daemons:
+                # It doesn't seem like mypy can figure out that rank
```

Comment:
it would be good to add some comment explaining what are we trying to achieve here :) ...  It seems that we are looking for a daemon withe maximum rank but I'm not sure.

Reply:
added a comment. What we're doing is getting the nfs daemon with highest rank generation for each rank. So if there is only one nfs of a given rank we just take that one, if there are multiple, we take the one with the higher rank_generation, in the end always ending up with one nfs for each rank of nfs we have deployed